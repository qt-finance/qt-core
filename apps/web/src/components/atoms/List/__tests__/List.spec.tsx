import React from 'react';
import { render } from '@testing-library/react';
import { composeStories } from '@storybook/testing-react';

import List from 'components/atoms/List';

import * as stories from './List.stories';

const testCases = Object.values(composeStories(stories)).map(Story => [
	// The ! is necessary in Typescript only, as the property is part of a partial type
	Story.storyName!, // eslint-disable-line @typescript-eslint/no-non-null-assertion
	Story,
]);
// Batch snapshot testing
test.each(testCases)('Renders %s story', async (_storyName, Story) => {
	const tree = await render(<Story />);
	expect(tree.baseElement).toMatchSnapshot();
});

const items = [
	{
		key: '1',
		value: 'Lee',
	},
	{
		key: '2',
		value: 'Mike',
	},
];

test('should have items property', () => {
	const { getByTestId } = render(<List items={items} />);

	expect(getByTestId('1')).toHaveTextContent('Lee');
});
