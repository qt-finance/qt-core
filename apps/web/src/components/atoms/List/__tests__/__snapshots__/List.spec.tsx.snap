// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Interactive story 1`] = `
<body>
  <div>
    <ul>
      <li
        data-testid="1"
      >
        Lee
      </li>
      <li
        data-testid="2"
      >
        Mike
      </li>
    </ul>
  </div>
</body>
`;
